scripts/colonycorefu.lua

```diff
@@ -3,76 +3,57 @@ require "/scripts/kheAA/transferUtil.lua"
 local bonusHappiness
 -- This is a modified well script. wellsDrawing is being used to represent how many tenants are active under this colony core.
 local rentTimer
-local rentGoal
+local rentTime
+local offlineTicks=0
+
+-- This object keep work even if it's unloaded. The trick is simple
+-- It simply remembers the time it was unloaded. When is loaded it will try to compensate all missed ticks.
+-- It will compensate maximum 24 hours of work.
 
 function init()
 	transferUtil.loadSelfContainer()
 	wellRange=config.getParameter("wellRange",128)
-	rentGoal=config.getParameter("productionTime",150)
+	rentTime=config.getParameter("productionTime",150)
+	wellSlots=config.getParameter('wellslots')
+	maxOfflineSeconds=config.getParameter("maxOfflineSeconds",86400)--86400 1 day
 	bonusHappiness = 10
 	rentTimer = 0
 	wellInit()
 	setDesc()
+
+	--figuring out how many ticks we missed
+	local leftTime = config.getParameter("leftTime")
+	if leftTime then
+		local diff=os.time() - leftTime
+		if diff>0 then
+			diff=math.min(math.floor(diff/rentTime),math.floor(maxOfflineSeconds/rentTime))
+			offlineTicks=diff-1
+			math.max(0, diff - 1)
```

Comment:
This does nothing. Should be above:
`offlineTicks=math.max(0, diff - 1)`

Reply:
Make sure to test (in-game) that everything works as expected:
1) when returning to the planet with Colony Core after 4-5- minutes have passed (which is more than `rentTime` seconds)
2) when returning to the planet immediately (before `rentTime` seconds: beam up, then return to the planet)

Reply:
> This does nothing. Should be above: `offlineTicks=math.max(0, diff - 1)`

That's why you should never code immediately after sleeping for 6 hours. Fixed this nonsense.

Reply:
Ok this additional check breaks our system. It causes our colony core to give you 1tick worth money EVERY time you beam on the planet (even if only 1 second passed). It happens because offlineTicks turns 0 and the next frame we add `(1+offlineTicks)`  ticks. Since offlineTicks=0, we add at least 1 tick every time we load our colony core.

So.. offline ticks should be -1 if we didn't wait enough time.

so the final check should be like this `offlineTicks=math.max(-1, diff - 1)`